src/drivers/intel/hda/hda-dma.c

```diff
@@ -221,18 +221,6 @@ static void hda_dma_get_dbg_vals(struct dma_chan_data *chan,
 #define hda_dma_ptr_trace(...)
```

Comment:
@keyonjie it took me several minutes to understand the commit message, and I think you are mixing L1 entry/exit.

suggested edit:

Exiting the DMA L1 state, e.g. on a timer tick when there is no data to copy, prevents the system from staying in S0ix power states. To extend S0ix residency, this L1 exit needs to happen only when copying data and the L1 mode can be re-entered in the post-copy stage.

Reply:
Thanks @plbossart . We don't have real L1 Entry:
In L1 Entry, we just increase the ref count host_dma_l1_sref.
In L1 Exit, we decrease the host_dma_l1_sref, and toggle the L1 Exit bit to force a L1 exit (flushing data from L1 Cache to DDR) if host_dma_l1_sref is 0.

So, the L1 exit does not happen when copying data, it should happen after those data are copied (that is, in the post-copy stage).

Reply:
@plbossart I take your suggestion and do a minor change according to what I explained above, updated now.